### DISCUSSION TITLE
TorchScript before TensorRT

### DISCUSSION THREAD

**Discussion by AndreiBaraian** [pytorch/TensorRT#1011] [Q&A] [answered]
Is it needed to compile a model to TorchScript via tracing/scripting before optimizing with TensortRT? 
I got this confusion by following the notebooks, where in the EfficientNet we directly optimize with TensorRT and do not use TorchScript. However, in the SSD notebook, it is mentioned that the model first need to be traced/scripted with TorchScript and only then TensorRT can be used. Could someone clarify this?

**Comment by narendasan** [comment, marked as answer]
So there were new APIs added in Torch-TensorRT 1.0 that allow you to provide a standard `nn.Module`. We will pass that through scripting for you then on to the compile function (there is no trace option for guaranteed correctness reasons) . The old API style now resides in `torch_tensorrt.ts`. You can also pass a torchscript module as to the new api if you like. 

Really its just syntax sugar to handle the most common use case but if you hit errors we still recommend manually tracing/scripting then passing it to Torch-TensorRT to determine where the source of the error is coming from.
